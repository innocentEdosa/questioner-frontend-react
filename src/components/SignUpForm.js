/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import formatServerError from '../helper/formatServerResponse';


const SignUpForm = ({
  blur,
  submit,
  values,
  changed,
  loading,
  serverError,
  onNavClick
}) => {
  const { formError } = values;

  const formattedServerError = formatServerError(serverError);


  return (
    <div className="page-bg">
      <div className=" form-wrapper">
        <form onSubmit={submit} className="form-signup needs-validation">
          <div className="text-center mb-4">
            <Link to="/">
              {' '}
              <img
                className="mb-4"
                src="assets/img/logo1.png"
                alt=""
                width="55"
                height="55"
              />
            </Link>
            <h1 className="h3 mb-3 font-weight-bold">Join Questioner</h1>
            <p className="text-muted">
              ...you could be the first to ask the right questions
            </p>
          </div>
          <div className="form-label-group">
            <label className="form-label" htmlFor="inputUsername">
              Username
            </label>
            <input
              onBlur={blur}
              onChange={changed}
              type="text"
              id="inputUsername"
              name="username"
              className="py-4 mb-3 form-control form-control-lg"
              placeholder="Pick a username"
              value={values.username}
              required
            />
            {formError.username ? (
              <p className="mt-n2 text-danger font-weight-lighter">
                {formError.username}
              </p>
            ) : null}
          </div>
          <div className="form-label-group">
            <label className="form-label" htmlFor="inputEmail">
              Email address
            </label>
            <input
              onBlur={blur}
              onChange={changed}
              type="email"
              id="inputEmail"
              name="email"
              value={values.email}
              className="py-4 mb-3 form-control form-control-lg"
              placeholder="You@example.com"
              required
            />
            {formError.email ? (
              <p className="mt-n2 text-danger font-weight-lighter">
                {formError.email}
              </p>
            ) : null}
          </div>
          <div className="form-row">
            <div className="col-md-6 ">
              <label className="form-label" htmlFor="inputPassword">
                Password
              </label>
              <input
                onBlur={blur}
                onChange={changed}
                type="password"
                name="password"
                id="inputPassword"
                className="py-4 mb-3 form-control form-control-lg"
                placeholder="Create a password"
                value={values.password}
                required
              />
              {formError.password ? (
                <p className="mt-n2 text-danger font-weight-lighter">
                  {formError.password}
                </p>
              ) : null}
            </div>
            <div className="col-md-6">
              <label className="form-label" htmlFor="confirmPassword">
                Confirm password
              </label>
              <input
                onBlur={blur}
                onChange={changed}
                type="password"
                id="confirmPassword"
                value={values.confirmPassword}
                name="password confirmation"
                className="py-4 mb-3 form-control form-control-lg"
                placeholder="Create a password"
                required
              />
              {formError['password confirmation'] ? (
                <p className="mt-n2 text-danger font-weight-lighter">
                  {formError['password confirmation']}
                </p>
              ) : null}
            </div>
          </div>
          {formattedServerError ? (
            <div className="alert alert-danger" role="alert">
              {formattedServerError}
            </div>
          ) : null}
          <button className="btn form-btn btn-lg btn-block " type="submit">
            {loading ? (
              <span className="spinner-border text-light" role="status" />
            ) : (
              'SIGN UP FOR QUESTIONER'
            )}
          </button>
          <p className="mt-5 mb-3 text-muted text-center">
            Already have an account?
            <Link to="/Auth" onClick={() => onNavClick('login')} className="ml-1 form-btn-sub btn btn-outline-success text-success ">
             Log in
              {' '}
            </Link>

          </p>
        </form>
      </div>
    </div>
  );
};

SignUpForm.propTypes = {
  blur: PropTypes.func.isRequired,
  submit: PropTypes.func.isRequired,
  changed: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  serverError: PropTypes.oneOf([null, PropTypes.object]).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  values: PropTypes.object.isRequired,
  onNavClick: PropTypes.func.isRequired
};
export default SignUpForm;
