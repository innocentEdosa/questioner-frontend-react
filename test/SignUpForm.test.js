import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
// import { shallow } from 'enzyme';
import SignUpForm from '../src/components/SignUpForm';

describe('<SignUpForm />', () => {
  it('Should match the snapshot', () => {
    const tree = renderer.create(<MemoryRouter><SignUpForm values={{ formError: { email: 'please provide an email' } }} /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
