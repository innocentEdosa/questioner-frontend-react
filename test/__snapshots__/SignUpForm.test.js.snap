// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> Should match the snapshot 1`] = `
<div
  className="page-bg"
>
  <div
    className=" form-wrapper"
  >
    <form
      className="form-signup needs-validation"
    >
      <div
        className="text-center mb-4"
      >
        <a
          href="/"
          onClick={[Function]}
        >
           
          <img
            alt=""
            className="mb-4"
            height="55"
            src="assets/img/logo1.png"
            width="55"
          />
        </a>
        <h1
          className="h3 mb-3 font-weight-bold"
        >
          Join Questioner
        </h1>
        <p
          className="text-muted"
        >
          ...you could be the first to ask the right questions
        </p>
      </div>
      <div
        className="form-label-group"
      >
        <label
          className="form-label"
          htmlFor="inputUsername"
        >
          Username
        </label>
        <input
          className="py-4 mb-3 form-control form-control-lg"
          id="inputUsername"
          name="username"
          placeholder="Pick a username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-label-group"
      >
        <label
          className="form-label"
          htmlFor="inputEmail"
        >
          Email address
        </label>
        <input
          className="py-4 mb-3 form-control form-control-lg"
          id="inputEmail"
          name="email"
          placeholder="You@example.com"
          required={true}
          type="email"
        />
        <p
          className="mt-n2 text-danger font-weight-lighter"
        >
          please provide an email
        </p>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-md-6 "
        >
          <label
            className="form-label"
            htmlFor="inputPassword"
          >
            Password
          </label>
          <input
            className="py-4 mb-3 form-control form-control-lg"
            id="inputPassword"
            name="password"
            placeholder="Create a password"
            required={true}
            type="password"
          />
        </div>
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
            htmlFor="confirmPassword"
          >
            Confirm password
          </label>
          <input
            className="py-4 mb-3 form-control form-control-lg"
            id="confirmPassword"
            name="password confirmation"
            placeholder="Create a password"
            required={true}
            type="password"
          />
        </div>
      </div>
      <button
        className="btn form-btn btn-lg btn-block "
        type="submit"
      >
        SIGN UP FOR QUESTIONER
      </button>
      <p
        className="mt-5 mb-3 text-muted text-center"
      >
        Already have an account?
        <a
          className="ml-1 form-btn-sub btn btn-outline-success text-success "
          href="/Auth"
          onClick={[Function]}
        >
          Log in
           
        </a>
      </p>
    </form>
  </div>
</div>
`;
